prince-of-versions/src/main/java/co/infinum/princeofversions/mvp/interactor/impl/PovInteractorImpl.java

```diff
@@ -25,22 +24,38 @@ public PovInteractorImpl(VersionVerifier versionVerifier, UpdateConfigLoader loa
     @Override
     public void checkForUpdates(final PovInteractorListener listener) {
         versionVerifier.verify(loader, new VersionVerifierListener() {
-            @Override
-            public void versionAvailable(VersionContext version) {
-                if (version.isCurrentLessThanMinimum()) {
-                    listener.onMandatoryUpdateAvailable(version);
-                } else if (version.hasOptionalUpdate() && version.isCurrentLessThanOptional()) {
-                    listener.onUpdateAvailable(version);
-                } else {
-                    listener.onNoUpdateAvailable(version);
-                }
-            }
+                    @Override
+                    public void versionAvailable(VersionContext version) {
+                        if (version.isCurrentLessThanMinimum()) {
+                            //If its a mandatory update, we check if the update has some optional parameters (sdk values in this case)
+                            if (!version.hasOptionalUpdate()) {
+                                //If it does not it means the update has no sdk restrictions set, to be more precise the restrictions
+                                //haven't changed which means the user's phone supports the minimum version of the app and thus we tell
+                                //listener to notify the user there's a new version of the app available
+                                listener.onMandatoryUpdateAvailable(version);
+                            } else {
+                                //If there are some optional parts we check if the user's phone supports the minimal version of the app
+                                //and its requirements
+                                if (version.getOptionalUpdate().getLastMinSdk() <= Build.VERSION.SDK_INT) {
```

Comment:
We need an interface which is used to fetch SDK VERSION. By using Build.VERSION.SDK_INT there is no way to mock SDK version in order to test different use-cases.